src/main/java/de/hsrm/vegetables/my_food_coop_service/controller/BalanceController.java

```diff
@@ -40,7 +97,10 @@
         UserDto userDto = userService.getUserById(userPrincipal.getId());
         userDto = userService.setBalance(userDto, request.getBalance());
 
-        return ResponseEntity.ok(Mapper.userDtoToBalanceResponse(userDto));
+        balanceService.saveBalanceChange(userDto, OffsetDateTime.now(), null,
```

Comment:
Done, `saveBalanceChange()` is called in `UserService` now. In case of `UserService::withdraw()` an extra parameter `saveBalanceChange` is passed to distinguish between a call from `BalanceController::balanceWithdraw()` and `PurchaseController::purchaseFromStock()`.